src/esplog/mini_decompressor.rs

```diff
@@ -0,0 +1,691 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use std::fmt::{Debug, Display};
+use std::ops::{Add, Div, Mul, Neg, Sub};
+
+#[derive(Copy, Clone, PartialOrd, PartialEq, Ord, Eq)]
+pub struct Fix32<const N: usize> {
+    v: i32,
```

Comment:
Just FYI, instead of named `v` parameter, Rust can do nameless parameters, like `pub struct Fix32<const N: usize>(i32)` and accessible by `.0` instead of `.v`.
It's fine either way, just letting you know